imap/annotate.c

```diff
@@ -1228,6 +1230,7 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
     }
 
```

Comment:
And then once we stop exiting early above, this if/else chain then needs to change shape too.  I think the three cases are something like:

* mbentry? => it's a single mailbox annotation
* pattern && *pattern? => it's a pattern
* otherwise => it's a server entry

In particular `else if (!*pattern) {` is bad now: we no longer get an empty string in the "pattern" argument during replication, because it's in the mboxname argument now instead, and pattern is NULL so *pattern segfaults.

I notice a few other invocations of `annotatemore_findall_pattern` (jmap, nntpd).  At a glance, two of them call it with an argument called "pattern", but the third calls it with an mbname_intname(mbname), and might also be a candidate for using the `annotatemore_findall_mboxname` version instead.

Reply:
Yep, patched the other annotatemore_findall_pattern too